### PR TITLE
Chapter 6

### PR DESCRIPTION
**Pull request by Moonforeva**

### CODE REVIEW

**Comment by Moonforeva** [comment]
Ready

File: Gemfile [hunk 1]
```diff
@@ -14,6 +14,7 @@ gem "config"
 gem "bootsnap", ">= 1.4.2", require: false
 gem "rails-i18n"
 gem "bootstrap-sass" 
+gem 'bcrypt', '~> 3.1', '>= 3.1.15'
```
Comment (saobv-2223) [inline Gemfile:17]: using ", please

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (saobv-2223) [inline app/models/user.rb:4]: move constant to top, and you should research gem config to manage the constant variable.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```
Comment (saobv-2223) [inline app/models/user.rb:2]: please, move under validate

**Comment by Moonforeva** [comment]
Ready

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    length: { maximum: Settings.validations.email.max_length},
+    format: { with: Settings.validations.email.regex },
+    uniqueness: true
+  
+  validates :password, presence: true, 
+    length: { minimum: Settings.validations.password.min_length }
+  
+  has_secure_password 
+  
+  before_save { self.email = email.downcase }
```
Comment (saobv-2223) [inline app/models/user.rb:15]:
```suggestion
  before_save :downcase_email
  
  private
  
  def downcase_email
    email.downcase!
  end
  
```

**Comment by Moonforeva** [comment]
Ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**